build-tools/packages/build-cli/src/commands/generate/changeset.ts

```diff
@@ -0,0 +1,263 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Package } from "@fluidframework/build-tools";
+import { VersionBumpType } from "@fluid-tools/version-tools";
+import { Flags } from "@oclif/core";
+import chalk from "chalk";
+import humanId from "human-id";
+import { writeFile } from "node:fs/promises";
+import path from "node:path";
+import prompts from "prompts";
+
+import { BaseCommand } from "../../base";
+import { Repository, getDefaultBumpTypeForBranch } from "../../lib";
+
+const DEFAULT_BRANCH = "main";
+const INSTRUCTIONS = `
+↑/↓: Change selection
+Space: Toggle selection
+a: Toggle all
+Enter: Done`;
+
+const excludedScopes = new Set(["@fluid-example", "@fluid-internal", "@fluid-test"]);
+
+interface Choice {
+	title: string;
+	value?: Package;
+	disabled?: boolean;
+	selected?: boolean;
+	heading?: boolean;
+}
+
+export default class GenerateChangesetCommand extends BaseCommand<typeof GenerateChangesetCommand> {
+	static summary = `Generates a new changeset file.`;
+	static aliases: string[] = ["changeset:add"];
+	static enableJsonFlag = true;
+
+	static flags = {
+		branch: Flags.string({
+			char: "b",
+			description:
+				"The branch to compare against. This is used to populate the list of changed packages.",
```

Comment:
Can we clarify what change/delta is in this context.

Reply:
I updated the wording here and elsewhere to hopefully address this. A package is considered changed if git reports any changes in its path when compared against a particular branch.